rllib/algorithms/ppo/ppo_rl_module_base.py

```diff
@@ -0,0 +1,90 @@
+"""
+This file holds framework-agnostic components for PPO's RLModules.
+"""
+
+import abc
+from typing import Mapping, Any
+
+import gymnasium as gym
+
+from ray.rllib.algorithms.ppo.ppo_catalog import PPOCatalog
+from ray.rllib.algorithms.ppo.ppo_rl_module_config import PPOModuleConfig
+from ray.rllib.core.rl_module.rl_module import RLModule, RLModuleConfig
+from ray.rllib.utils.annotations import override, ExperimentalAPI
+from ray.rllib.utils.gym import convert_old_gym_space_to_gymnasium_space
+from ray.rllib.core.models.base import ActorCriticEncoder
+
+
+@ExperimentalAPI
+class PPORLModuleBase(RLModule, abc.ABC):
+    framework = None
+
+    def __init__(self, config: RLModuleConfig):
+        super().__init__()
+        self.config = config
+        catalog = config.catalog
+
+        assert isinstance(catalog, PPOCatalog), "A PPOCatalog is required for PPO."
+
+        # Set input- and output dimensions to fit PPO's needs.
+        catalog.encoder_config.input_dim = self.config.observation_space.shape[0]
```

Comment:
Shouldn't these be set already as part of constructing the catalog?

Reply:
This constructor should simply only include the catalog.build_xxxx(fw) calls.

Reply:
Yep, you are right.